puresasl/mechanisms.py

```diff
@@ -187,6 +190,56 @@ def process(self, challenge=None):
     def dispose(self):
         self.password = None
 
+    def wrap(self, outgoing):
+        if self.qop == 'auth-int' or self.qop == 'auth-conf':
+            raise Exception('{0} QoP not supported for CRAM-MD5'.format(self.qop))
```

Comment:
Yes, you're right, it should be checked up front. Checking in wrap/unwrap should only be necessary if supporting multiple QOPs.